server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/MailboxGetMethod.scala

```diff
@@ -0,0 +1,89 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.method
+
+import eu.timepit.refined.auto._
+import javax.inject.Inject
+import org.apache.james.jmap.json.Serializer
+import org.apache.james.jmap.mail._
+import org.apache.james.jmap.model.Invocation.{Arguments, MethodName}
+import org.apache.james.jmap.model.State.INSTANCE
+import org.apache.james.jmap.model.{Invocation, MailboxFactory}
+import org.apache.james.jmap.utils.quotas.QuotaLoaderWithPreloadedDefaultFactory
+import org.apache.james.mailbox.model.MailboxMetaData
+import org.apache.james.mailbox.model.search.MailboxQuery
+import org.apache.james.mailbox.{MailboxManager, MailboxSession}
+import org.apache.james.metrics.api.MetricFactory
+import org.reactivestreams.Publisher
+import play.api.libs.json.{JsError, JsObject, JsSuccess}
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+class MailboxGetMethod @Inject() (serializer: Serializer,
+                                  mailboxManager: MailboxManager,
+                                  quotaFactory : QuotaLoaderWithPreloadedDefaultFactory,
+                                  mailboxFactory: MailboxFactory,
+                                  metricFactory: MetricFactory) extends Method {
+  override val methodName: MethodName = MethodName("Mailbox/get")
+
+  override def process(invocation: Invocation, mailboxSession: MailboxSession): Publisher[Invocation] = {
+    metricFactory.runPublishingTimerMetricLogP99(JMAP_RFC8621_PREFIX + methodName.value,
+      asMailboxGetRequest(invocation.arguments)
+        .flatMap(mailboxGetRequest => getMailboxes(mailboxGetRequest, mailboxSession)
+          .collectSeq()
```

Comment:
Still no sortOrder sort?

Do we have integration tests about it (at least?)

Reply:
There is one for get all mailboxes... It works because I saw (don't remember where exactly) a sort being done more done the line. Pretty sure it's gonna fail for get mailboxes by ids though

I couldn't solve the sorting with SFlux / SMono so far in scala... but I can try to check again